### PR TITLE
vlek: Adding VLEK support

### PR DESCRIPTION
**Pull request by larrydewey**
This adds support for handling Version Loaded Endorsement Keys and Certificate Revocation Lists.

### CODE REVIEW

File: src/certs/snp/chain.rs [hunk 1]
```diff
@@ -97,32 +105,39 @@ impl Chain {
...
-            return Err(Error::new(other, "VCEK not found"));
+        } else if vcek.is_none() && vlek.is_none() {
+            return Err(Error::new(other, "VCEK/VLEK not found"));
         }
 
+        // Use the VLEK whenever it is present, but use the VCEK when VLEK is missing.
+        let vek_val = match (vcek, vlek) {
+            (_, Some(vlek)) => vlek,
+            (Some(vcek), None) => vcek,
+            _ => unreachable!(),
```
Comment (tylerfanelli) [inline src/certs/snp/chain.rs:116]: `(Some(vcek), Some(vlek)`?

Comment (larrydewey) [reply src/certs/snp/chain.rs:116]: That is covered on line 114

Comment (tylerfanelli) [reply src/certs/snp/chain.rs:116]: I misread what that meant. Thanks! LGTM.

**Review by tylerfanelli — APPROVED** [review]

**Review by DGonzalezVillal — APPROVED** [review]
lgtm

**Pull request merged**